spec/models/game_spec.rb

```diff
@@ -10,89 +10,28 @@ def game_with_input_sequence(input)
     end
     game
   end
-  
-
-  it "handles a valid input sequence" do
-    input = %w(l l r r l l)
-    game = game_with_input_sequence(input)
-    expect(game.p1_score).to eq(4)
-    expect(game.p2_score).to eq(2)
-    expect(game.set).to eq(1)
-  end
-
-  it "can be rerun" do
-    input = %w(l l r r l l)
-    game = game_with_input_sequence(input)
-    expect(game.p1_score).to eq(4)
-    expect(game.p2_score).to eq(2)
-    expect(game.set).to eq(1)
-    %w(l r).each do |c| 
-      game.handle_input(c)
-    end
-    expect(game.p1_score).to eq(5)
-    expect(game.p2_score).to eq(3)
-  end
-
-  it "handles changeover" do
-    input = %w(l) * 13
-    game = game_with_input_sequence(input)
-    expect(game.p1_score).to eq(0)
-    expect(game.p2_score).to eq(1)
-    expect(game.p1_set_score).to eq(1)
-    expect(game.set).to eq(2)
 
-  end
-
-  it "knows who won" do
-    #after 3*1b -r we are at 10-0 in set 3
-    input = %w(l)*12 + %w(r)*12 + %w(l)*10
-    game = game_with_input_sequence(input)
-    expect(game.winner).to be_nil
-    game.handle_input('l')
-    expect(game.winner).to eq(1)
-  end
-
-  it "is aware of minimum difference" do
-    #input = %w(1) * (3*11-1)
+  it "can delegate input to score" do
+    score = game_with_input_sequence(%w(l l r r l l)).score
+    expect(score.p1_score).to eq(4)
+    expect(score.p2_score).to eq(2)
+    expect(score.set).to eq(1)
   end
 
   it "can undo" do 
     game = game_with_input_sequence(%w(l l r r l))
-    expect(game.p1_score).to eq(3)
+    expect(game.score.p1_score).to eq(3)
```

Comment:
Just an idea – maybe these “undo”-tests should look like this:
```ruby
game = game_with_input_sequence(%w(l l r r l))
original_score = game.score
game.handle_input('r')
expect(game.score).to_not eq(original_score)
game.undo
expect(game.score).to eq(original_score)
```

Of course we’d need to implement `Score#==` for this to work.